llvm/include/llvm/CodeGen/GlobalISel/GIMatchTableExecutor.h

```diff
@@ -275,7 +275,7 @@ enum {
   /// Check the operand is a specific intrinsic ID
   /// - InsnID(ULEB128) - Instruction ID
   /// - OpIdx(ULEB128) - Operand index
-  /// - IID(2) - Expected Intrinsic ID
```

Comment:
Could you omit the extra 2 bytes by assuming the current target's index?

Reply:
I wasn't sure. What if a GISel patterns refers to a generic intrinsic or some other targets intrinsics? For instance, AMDGPU likely refers to mix of r600 and amdgcn intrinsics?

Reply:
AMDGPU shouldn't mix r600 and amdgcn intrinsics, but there might be a handful remaining that do that mixing. @AlexVlx also wanted to use amdgcn intrinsics in spirv

Reply:
> AMDGPU shouldn't mix r600 and amdgcn intrinsics, but there might be a handful remaining that do that mixing. @AlexVlx also wanted to use amdgcn intrinsics in spirv

We do use them for AMDGCN flavoured SPIR-V, but I admit I was not aware and have not really internalised what this change would imply in that context. I’ll try to parse it now.

Reply:
Thanks. I don't think it will directly affect anything at the moment. A follow on change would need to be aware of which targets use which intrinsic prefixes to make sure we do disable the right ones for that target,

Reply:
My concern with making this 2 byte is: (1) won't we have files with patterns that use mix of target independent and target dependent builtins. In such cases, just encoding the 16-bit target index is not sufficient, and  (2) premature optimizations? If its really important, I'd prefer doing this as an independent change.